File: JavaScript-Programs/factorial.js
//program to find factorial of a number
// input : 4 :: 4 x 3 x 2 x 1
//input : 0 :: o/p --> 1

let factorial = (num) => {
  // ! check if number is 0 , then return 1 if true
  if (num === 0) {
    return 1;
  } else {
    // * if positive, run a recurive function
    return num * factorial(num - 1);
  }
};

//o/p
console.log(factorial(4)); // 4 x 3 x 2 x 1 = 24

File: Java-Programs/Factorial.java
import java.util.Scanner;
public class Factorial {
	public static void main(String[] args) {
		System.out.print("Enter a number: ");
		Scanner input=new Scanner(System.in);
    int num=input.nextInt();
		long factorial=1;
		for (int i =1; i<=num; ++i) 
			factorial*=i;
		
		System.out.printf("Factorial of %d = %d", num, factorial);
	}
}
